data/src/test/java/org/apache/iceberg/TestSplitScan.java

```diff
@@ -32,59 +35,50 @@
 import org.apache.iceberg.io.FileAppender;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.apache.iceberg.types.Types;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-
-@RunWith(Parameterized.class)
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+@ExtendWith(ParameterizedTestExtension.class)
 public class TestSplitScan {
   private static final Configuration CONF = new Configuration();
   private static final HadoopTables TABLES = new HadoopTables(CONF);
-
   private static final long SPLIT_SIZE = 16 * 1024 * 1024;
 
   private static final Schema SCHEMA =
       new Schema(
           required(1, "id", Types.IntegerType.get()), required(2, "data", Types.StringType.get()));
 
+  @Parameters(name = "fileFormat = {0}")
+  public static Collection<FileFormat> parameters() {
+    return Arrays.asList(FileFormat.PARQUET, FileFormat.AVRO);
+  }
+
   private Table table;
   private File tableLocation;
-
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
   private List<Record> expectedRecords;
 
-  @Parameterized.Parameters(name = "format = {0}")
-  public static Object[] parameters() {
-    return new Object[] {"parquet", "avro"};
-  }
+  @Parameter private FileFormat format;
+  @TempDir private File tempDir;
 
-  private final FileFormat format;
-
-  public TestSplitScan(String format) {
-    this.format = FileFormat.fromString(format);
-  }
-
-  @Before
+  @BeforeEach
   public void before() throws IOException {
-    tableLocation = new File(temp.newFolder(), "table");
+    tableLocation = new File(tempDir, "table");
     setupTable();
   }
 
-  @Test
+  @TestTemplate
   public void test() {
-    Assert.assertEquals(
-        "There should be 4 tasks created since file size is approximately close to 64MB and split size 16MB",
-        4,
-        Lists.newArrayList(table.newScan().planTasks()).size());
+    List<CombinedScanTask> tasks = Lists.newArrayList(table.newScan().planTasks());
+    assertThat(tasks)
+        .as(
+            "There should be 4 tasks created since file size is approximately close to 64MB and split size 16MB")
+        .hasSize(4);
+
     List<Record> records = Lists.newArrayList(IcebergGenerics.read(table).build());
-    Assert.assertEquals(expectedRecords.size(), records.size());
-    for (int i = 0; i < expectedRecords.size(); i++) {
-      Assert.assertEquals(expectedRecords.get(i), records.get(i));
-    }
+    assertThat(records).hasSize(expectedRecords.size());
```

Comment:
```suggestion
    assertThat(records).hasSameSizeAs(expectedRecords).isEqualTo(expectedRecords);
```
both assertions can be combined to a single one

Reply:
yep, done